website/docs/resolving/method-injection.mdx

```diff
@@ -20,14 +18,14 @@ public class SomeBehaviour : MonoBehaviour
 }
 ```
 
-:::tip Recommendation
-Consider whether injection to MonoBehaviour is necessary.
-In a code base where domain logic and presentation are well decoupled, MonoBehaviour should act as a View component.
+The `[Inject]`-annotated method can have any name and any accessibility level.
 
-In my opinion, View components should only be responsible for rendering and should be flexible.
+If you inject dependencies into `MonoBehaviour`s, you must be mindful of when the `[Inject]`-annotated method is called with respect to [Unity's object lifecycle](https://docs.unity3d.com/Manual/ExecutionOrder). If your [`Awake()`](https://docs.unity3d.com/ScriptReference/MonoBehaviour.Awake), [`Start()`](https://docs.unity3d.com/ScriptReference/MonoBehaviour.Start), or [`OnEnable()`](https://docs.unity3d.com/ScriptReference/MonoBehaviour.OnEnable) methods need any dependencies from VContainer, consider moving their logic into your injection method.
```

Comment:
This is true. However, I do not want to recommend writing logic inside Inject methods.

Actually, Im working on getting the Inject method to execute earlier than Awake. :p
( Perhaps that would be preferable.

For example, the following
https://github.com/hadashiA/VContainer/blob/master/VContainer/Assets/VContainer/Runtime/Unity/PrefabComponentRegistration.cs#L48

So, I'd like to weaken the wording of this, or put it a bit further down in `:::tip ~ :::`. how about this ?

Reply:
I thought that `Awake()` methods were called as soon as the owning `GameObject` is instantiated, regardless of whether or not the `GameObject` itself is enabled?

Reply:
What would you suggest instead of moving initialization to `[Inject]` methods (both for right now and for when you implement the changes you're working on)?

Reply:
Oh, and this might be important; the reason I suggest moving initialization to `[Inject]` is that in my own project I spawn almost all prefabs using Addressables, so I can't use `IObjectResolver.Instantiate` (or whatever the extension method is called).

Reply:
>  I thought that Awake() methods were called as soon as the owning GameObject is instantiated, regardless of whether or not the GameObject itself is enabled?

There is documentation on this:
>  Awake: This function is always called before any Start functions and also just after a prefab
is instantiated. (If a GameObject is inactive during start up Awake is not called until it is made active.)

https://docs.unity3d.com/Manual/ExecutionOrder.html
 

>  What would you suggest instead of moving initialization to [Inject] methods (both for right now and for when you implement the changes you're working on)?

I think that `Start()` is fine for most processes.
( As you said, Awake / OnEnable may be a timing issue. However, I think Start can be controlled to be called later than Inject.

In fact, I don't see a problem with doing a lot of things in the Inject method.

On the other hand, I think the constructor of an object should avoid causing side effects to other objects or executing operations that cause exceptions.

For the [Inject] method, I prefer to keep it to a role almost similar to that of a constructor.

Reply:
> 
> 
> > I thought that Awake() methods were called as soon as the owning GameObject is instantiated, regardless of whether or not the GameObject itself is enabled?
> 
> There is documentation on this:
> 
> > Awake: This function is always called before any Start functions and also just after a prefab
> > is instantiated. (If a GameObject is inactive during start up Awake is not called until it is made active.)
> 
> https://docs.unity3d.com/Manual/ExecutionOrder.html

Neat. I might submit another revision that explicitly refers to this, then.

> > What would you suggest instead of moving initialization to [Inject] methods (both for right now and for when you implement the changes you're working on)?
> 
> I think that `Start()` is fine for most processes.
> ( As you said, Awake / OnEnable may be a timing issue. However, I think Start can be controlled to be called later than Inject.
> 
> In fact, I don't see a problem with doing a lot of things in the Inject method.

Neither do I, which is why I asked. But it's nice to know that it's not the only option.

> On the other hand, I think the constructor of an object should avoid causing side effects to other objects or executing operations that cause exceptions.
> 
> For the [Inject] method, I prefer to keep it to a role almost similar to that of a constructor.

True, but I consider that more of a (strong) guideline. Sometimes you have to bend the rules so they don't break.